File: FloatItWatch/src/main/java/net/geekstools/floatshort/PRO/Utils/OpenApplications.java
/*
 * Copyright © 2020 By Geeks Empire.
 *
 * Created by Elias Fazel
 * Last modified 4/25/20 11:54 AM
 *
 * Licensed Under MIT License.
 * https://opensource.org/licenses/MIT
 */

package net.geekstools.floatshort.PRO.Utils;

import android.app.Activity;
import android.os.Bundle;

import net.geekstools.floatshort.PRO.Utils.Functions.FunctionsClass;

public class OpenApplications extends Activity {

    FunctionsClass functionsClass;

    @Override
    protected void onCreate(Bundle Saved) {
        super.onCreate(Saved);
        functionsClass = new FunctionsClass(getApplicationContext());
        String packageName = getIntent().getStringExtra("packageName");
        try {
            if (getIntent().hasExtra("className")) {
                String className = getIntent().getStringExtra("className");
                functionsClass.openApplication(packageName, className);
            } else {
                functionsClass.openApplication(packageName);
            }
        } catch (Exception e) {
            e.printStackTrace();
            functionsClass.openApplication(packageName);
        }
        finish();
    }
}

File: FloatItWatch/src/main/java/net/geekstools/floatshort/PRO/Utils/LaunchPad/OpenApplications.kt
/*
 * Copyright © 2020 By Geeks Empire.
 *
 * Created by Elias Fazel
 * Last modified 4/26/20 8:26 AM
 *
 * Licensed Under MIT License.
 * https://opensource.org/licenses/MIT
 */
package net.geekstools.floatshort.PRO.Utils.LaunchPad

import android.app.Activity
import android.os.Bundle
import net.geekstools.floatshort.PRO.Utils.Functions.FunctionsClass

class OpenApplications : Activity() {

    override fun onCreate(Saved: Bundle?) {
        super.onCreate(Saved)

        val functionsClass = FunctionsClass(applicationContext)

        val packageName = intent.getStringExtra("packageName")

        if (intent.hasExtra("className")) {
            val className = intent.getStringExtra("className")

            functionsClass.openApplication(packageName, className)

        } else {

            functionsClass.openApplication(packageName)
        }
        try {

        } catch (e: Exception) {
            e.printStackTrace()

            functionsClass.openApplication(packageName)
        }

        this@OpenApplications.finish()
    }
}